src/executor/stack/mod.rs

```diff
@@ -428,22 +467,22 @@ impl<'config, S: StackState<'config>> StackExecutor<'config, S> {
 		}
 	}
 
-	fn call_inner(
+	pub fn call_prepare(
```

Comment:
Is it possible to make `create_prepare` and `call_prepare` take only `&self` and not mut?

Reply:
Unfortunately no, they both need mutable access to the `state` and `metadata`.